### PR TITLE
Add OAuth support to /institutions endpoints

### PR DESCRIPTION
**Pull request by jacksiegel**
Adds support for a new `oauth` request option for `/institutions/get` and `/institutions/search`, and adds a new `oauth` field to the institution schema returned by `/institutions/get`, `/institutions/get_by_id`, and `/institutions/search`.

### CODE REVIEW

File: src/test/java/com/plaid/client/integration/InstitutionsGetTest.java [hunk 1]
```diff
@@ -70,6 +71,18 @@ public void testSuccessWithCountryCodes() throws Exception {
     assertEquals(3, response.body().getInstitutions().size());
   }
 
+  @Test
+  public void testSuccessWithOAuth() throws Exception {
+    Response<InstitutionsGetResponse> response =
+      client().service().institutionsGet(new InstitutionsGetRequest(3, 0).withCountryCodes(Arrays.asList("GB")).withOAuth()).execute();
```
Comment (notthefakestephen) [inline src/test/java/com/plaid/client/integration/InstitutionsGetTest.java]: For the other tests in the file, can we also add an `assertFalse(institutions.get(i).getOAuth());` to make sure we aren't returning any `oauth` institutions  when we should not be?

File: src/test/java/com/plaid/client/integration/InstitutionsSearchTest.java [hunk 2]
```diff
@@ -79,6 +79,15 @@ public void testSuccessWithAccountFilter() throws Exception {
     assertTrue(response.body().getInstitutions().size() > 0);
   }
 
+  @Test
+  public void testSuccessWithOAuth() throws Exception {
+    Response<InstitutionsSearchResponse> response =
+      client().service().institutionsSearch(new InstitutionsSearchRequest("Flexible Platypus Open Banking").withCountryCodes(Arrays.asList("GB")).withOAuth()).execute();
+    assertSuccessResponse(response);
+    InstitutionsSearchResponse institutionsSearchResponse = response.body();
+    assertTrue(institutionsSearchResponse.getInstitutions().get(0).getOAuth());
```
Comment (notthefakestephen) [inline src/test/java/com/plaid/client/integration/InstitutionsSearchTest.java]: same comment here https://github.com/plaid/plaid-java/pull/197/files#r384532004

File: src/test/java/com/plaid/client/integration/InstitutionsGetTest.java (hunk 1 above)
Comment (jacksiegel) [reply src/test/java/com/plaid/client/integration/InstitutionsGetTest.java]: ⚡️ daa9949

File: src/test/java/com/plaid/client/integration/InstitutionsSearchTest.java (hunk 2 above)
Comment (jacksiegel) [reply src/test/java/com/plaid/client/integration/InstitutionsSearchTest.java]: ⚡️ daa9949

**Pull request merged**